src/components/Comment/Comment.jsx

```diff
@@ -0,0 +1,172 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import Textarea from 'react-textarea-autosize';
+import { PropTypes } from 'prop-types';
+import ContentLoader from 'react-content-loader';
+import {
+  handleCommentInput,
+  submitComment,
+  deleteComment,
+  updateComment,
+  setUpdateCommentBody,
+} from '../../redux/actions/commentAction';
+import CommentRender from './CommentRender';
+
+export class Comment extends Component {
+  addComment = () => {
+    const { body, articleSlug, onSubmitComment } = this.props;
+    onSubmitComment(body, articleSlug);
+  }
+
+  onChange = (e) => {
+    const { onCommentInput } = this.props;
+    onCommentInput({ body: e.target.value });
+    e.preventDefault();
+  }
+
+  displayComments = () => {
+    const {
+      commentList,
+      currentUser,
+      onDeleteComment,
+      articleSlug,
+      onUpdateCommentInput,
+      fetching,
+    } = this.props;
+    if (!commentList.length && fetching) {
+      return (
+        <ContentLoader
+          height={160}
+          width={400}
+          speed={2}
+          primaryColor="#f3f3f3"
+          secondaryColor="#ecebeb"
+        >
+          <rect x="0" y="80" rx="3" ry="3" width="350" height="6" />
+          <rect x="0" y="95" rx="3" ry="3" width="380" height="6" />
+          <rect x="0" y="110" rx="3" ry="3" width="201" height="6" />
+          <circle cx="22" cy="50" r="21" />
+        </ContentLoader>
+      );
+    }
+    const newList = commentList.sort((a, b) => new Date(b.createdAt) - new Date(a.createdAt));
+    return (
+      newList.map(comment => (
+        <CommentRender
+          commentList={commentList}
+          currentUser={currentUser}
+          onDeleteComment={onDeleteComment}
+          articleSlug={articleSlug}
+          comment={comment}
+          key={comment.id}
+          enterPress={this.onEnterPress}
+          updateComment={this.onEditComment}
+          inputHandler={onUpdateCommentInput}
+        />
+      ))
+    );
+  }
+
+  commentForm = () => {
+    const { body } = this.props;
+    return (
+      <form>
+        <Textarea
+          className="comment-textarea new"
+          placeholder="Add your comment..."
+          type="text"
+          value={body}
+          onChange={this.onChange}
+          onKeyDown={e => this.onEnterPress(e, this.addComment)}
+          data-el="comment-input"
+        />
+      </form>
+    );
+  }
+
+  onEnterPress = (e, fun, id) => {
```

Comment:
What is fun supposed to represent here? Always name your variables in a way other people can understand what they represent.

Reply:
fun stands for a function. I couldn't find any other variable name to use. Could suggest one?